lattices/LatticeMath/LatticeStatistics.tcc

```diff
@@ -928,98 +1222,95 @@ void LatticeStatistics<T>::_doStatsLoop(
             // I don't understand, things seem to work OK when I try this in tIPosition, but not here.
             _chauvIters[os.str()] = ch->getNiter();
         }
-        // FIXME it is likely that these putAt() calls are the source of the performance
-        // degradation in the case where the old method is faster. Eg consider a lattice
-        // with shape 1000 x 1000 x 3 doing stats with a cursor axis of 2 so that 1000 x 1000
-        // sets of stats are computed. That's 1e6 putAt calls for *each* statistic. A relatively
-        // small array that holds many statistics sets should probably be used as temporary storage,
-        // and when the array is full it should be inserted into the storage lattice using
-        // putSlice(). This is how the old method manages puts, and probably is why it is
-        // so much faster in these types of cases.
-        pStoreLattice_p->putAt(stats.mean, posMean);
-        pStoreLattice_p->putAt(stats.npts, posNpts);
-        pStoreLattice_p->putAt(stats.sum, posSum);
-        pStoreLattice_p->putAt(stats.sumsq, posSumsq);
-        pStoreLattice_p->putAt(stats.variance, posVariance);
-        if (fixedMinMax_p && ! noInclude_p) {
-            currentMax = range_p[1];
+        if (isReal && (! fixedMinMax_p || noInclude_p)) {
+            IPosition maxPos, minPos;
+            Bool atStart = stepper.atStart();
+            if (atStart || currentMin < overallMin || currentMax > overallMax) {
+                dataProvider->minMaxPos(minPos, maxPos);
+                _updateMinMaxPos(
+                    overallMin, overallMax, currentMin, currentMax,
+                    minPos, maxPos, atStart, subLat
+                );
+            }
         }
-        else {
-            currentMax = stats.max.null() ? 0 : *stats.max;
+        if(! progressMeter.null() && nsetsIsLarge) {
+            (*progressMeter)++;
         }
-        pStoreLattice_p->putAt(currentMax, posMax);
-        if (fixedMinMax_p && ! noInclude_p) {
-            currentMin = range_p[0];
+    }
+}
+
+template <class T>
+void LatticeStatistics<T>::_updateMinMaxPos(
+    T& overallMin, T& overallMax, T currentMin, T currentMax,
+    const IPosition& minPos, const IPosition& maxPos,
+    Bool atStart, const SubLattice<T>& subLat
+) {
+    // CAUTION The way this has worked in the past apparently for
+    // lattices is that the max and min positions are representative
+    // of the *entire* lattice, and were not stored on a sublattice
+    // by sublattice basis. This is easy to fix now,
+    // but for backward compatibility, I'm leaving this functionality as
+    // it has been.
```

Comment:
Do you see any way forward, to move away from the historical semantics? Your description implies that the historical version is not what it should be.

Reply:
It really depends on what users want. So far they've not complained about this, and I don't want to change it for backward compatibility purposes. In the end, it seems to be an issue of minor concern. If they want these per lattice, it will require removing the original code, which we still use, because in some cases, it performs better than the newer code.